InternalDocs/garbage_collector.md

```diff
@@ -0,0 +1,596 @@
+
+Garbage collector design
+========================
+
+Abstract
+========
+
+The main garbage collection algorithm used by CPython is reference counting. The basic idea is
+that CPython counts how many different places there are that have a reference to an
+object. Such a place could be another object, or a global (or static) C variable, or
+a local variable in some C function. When an object’s reference count becomes zero,
+the object is deallocated. If it contains references to other objects, their
+reference counts are decremented. Those other objects may be deallocated in turn, if
+this decrement makes their reference count become zero, and so on. The reference
+count field can be examined using the ``sys.getrefcount`` function (notice that the
+value returned by this function is always 1 more as the function also has a reference
+to the object when called):
+
+```
+    >>> x = object()
+    >>> sys.getrefcount(x)
+    2
+    >>> y = x
+    >>> sys.getrefcount(x)
+    3
+    >>> del y
+    >>> sys.getrefcount(x)
+    2
+```
+
+The main problem with the reference counting scheme is that it does not handle reference
+cycles. For instance, consider this code:
+
+```
+    >>> container = []
+    >>> container.append(container)
+    >>> sys.getrefcount(container)
+    3
+    >>> del container
+```
+
+In this example, ``container`` holds a reference to itself, so even when we remove
```

Comment:
```suggestion
In this example, `container` holds a reference to itself, so even when we remove
```
sphinx->md

Reply:
Probably not worth doing these replacements? ``` `` ``` works fine in markdown for literal syntax (you can have an arbitrary number of `` ` ``).